### ISSUE TITLE
Load kernel modules before UpdateInterface

### ISSUE THREAD

**Issue by xiaosuiba** [kata-containers/kata-containers#4247]
**Is your feature request related to a problem? Please describe.**

When using kata with SRIOV type cni, sometimes the nic on guest os needs to be provisioned by a kernel module(e.g. mlx5_core). Kata-agent does an UpdateInterface before load kernel modules and this will fail because the kernel modules needed by the sriov device haven't been load yet, thus kata-agent cannot find the network devices.

The code related sits [here](https://github.com/kata-containers/kata-containers/blob/main/src/runtime/virtcontainers/kata_agent.go#L732)

```golang
	if err = k.updateInterfaces(ctx, interfaces); err != nil {
		return err
	}
	if _, err = k.updateRoutes(ctx, routes); err != nil {
		return err
	}
	if err = k.addARPNeighbors(ctx, neighs); err != nil {
		return err
	}

	storages := setupStorages(ctx, sandbox)

	kmodules := setupKernelModules(k.kmodules)

	req := &grpc.CreateSandboxRequest{
...
``` 

**Describe the solution you'd like**

Load kernel modules before UpdateInterface

**Describe alternatives you've considered**

We added a `ExecStartPre=modprobe mlx5_core` in kata-agent.service right now as a workaround.

**Additional context**

#4091

**Comment by bergwolf** [comment]
Yes, indeed we should create sandbox before setting up the network devices in case any extra kernel modules are required by the network device. @xiaosuiba would you like to submit a fix?

**Comment by fu7100** [comment]
I have encountered a problem. After Kata container is allocated to Kata pod using the sriov VF interface, the container cannot be started. Is it related to this?

**Comment by xujunjie-cover** [comment]
I think rebuild kernel will be better to reslov it.

**Comment by fu7100** [comment]
@xujunjie-cover  I have compiled mlx5_core into the kernel, so do I still need to execute the `modprobe mlx5_core` command?

**Comment by xujunjie-cover** [comment]
> @xujunjie-cover I have compiled mlx5_core into the kernel, so do I still need to execute the `modprobe mlx5_core` command?

no need.
